pkg/hive/apiserver/registry/util/convert.go

```diff
@@ -157,13 +164,21 @@ func ClusterDeploymentToHiveV1(in *hiveapi.ClusterDeployment, sshKey string, out
 }
 
 // ClusterDeploymentFromHiveV1 turns a v1 ClusterDeployment into a v1alpha1 ClusterDeployment.
-func ClusterDeploymentFromHiveV1(in *hivev1.ClusterDeployment, installConfig *installtypes.InstallConfig, machinePools []*hivev1.MachinePool, out *hiveapi.ClusterDeployment) error {
+func ClusterDeploymentFromHiveV1(in *hivev1.ClusterDeployment, installConfig *installtypes.InstallConfig, machinePools []*hivev1.MachinePool, installConfigResourceVersion string, out *hiveapi.ClusterDeployment) error {
 	in = in.DeepCopy()
 
 	if err := hiveconversion.Convert_v1_ClusterDeployment_To_v1alpha1_ClusterDeployment(in, out, nil); err != nil {
 		return err
 	}
 
+	resourceVersions := make([]string, 2+len(machinePools))
+	resourceVersions[0] = in.ResourceVersion
+	resourceVersions[1] = installConfigResourceVersion
```

Comment:
should you have a place holder for when the installConfigResourceVersion is blank?

Reply:
The placeholder is the empty string. What would we get out of a non-empty placeholder?

Reply:
Just less confusion when you examine a resource and it has `aaa--bbb` instead of `aaa-NONE-bbb` (the latter is more explicit imho)

Reply:
My preference would be to leave it as an empty string.

1) There should rarely if ever be a case where there is not an install-config secret.
2) The resource version is not something that should usually have to be inspected by a user.
3) This would require additional code in the creation of the resource version as well as the parsing of the resource version for its constituent parts.

Reply:
👍